library/src/main/java/org/antennapod/audio/SonicAudioPlayer.java

```diff
@@ -777,7 +838,10 @@ public void run() {
                 }
                 Log.d(TAG_TRACK, "Decoding loop exited. Stopping codec and track");
                 Log.d(TAG_TRACK, "Duration: " + (int) (mDuration / 1000));
-                Log.d(TAG_TRACK, "Current position: " + (int) (mExtractor.getSampleTime() / 1000));
+
+                if (!((mInitiatingCount.get() > 0) || (mSeekingCount.get() > 0))) {
```

Comment:
could you maybe extract `((mInitiatingCount.get() > 0) || (mSeekingCount.get() > 0))` into a method describing its meaning?
goal: `if ([!]someThingIsTheCase()) { ...`